packages/components/src/components/Progress/ProgressItem.tsx

```diff
@@ -0,0 +1,232 @@
+     */
+    onCopyTextButtonPress?: (message: ProgressTaskErrorDetails) => void;
+    /**
+     * Callback function that is invoked when the retry button is pressed, allowing the specific failed task to be retried.
+     * @param task The task object that failed and needs to be retried.
+     */
+    onRetryButtonPress?: (task: ProgressTask) => void;
+} & ProgressItemPropsOptions;
+
+export const ProgressItem = ({
+    title,
+    description,
+    showCopyTextButton,
+    showRetryButton,
+    status = "notStarted",
+    tasks = [],
+    onCopyTextButtonPress,
+    onRetryButtonPress,
+}: ProgressItemProps) => {
+    const [isExpanded, setIsExpanded] = useState(false);
+
+    const token = useToken();
+    const styles = useStyles(themeStyles);
+
+    const taskCounter = tasks?.length;
+    const completedTaskCounter = tasks?.filter(task => task.status === "success").length;
+
+    const animationConfig = {
+        duration: token.timing.animation.slow,
+        create: {
+            type: LayoutAnimation.Types.easeInEaseOut,
+            property: LayoutAnimation.Properties.opacity,
+        },
+        update: {
+            type: LayoutAnimation.Types.easeInEaseOut,
+        },
+    };
+
+    const taskStatus = computeTaskStatus(tasks, status);
+    const taskHasError = taskStatus === "error";
+    const failedTask = tasks.find(task => task.status === "error");
+
+    const handleCopyTextButtonPress = () => {
+        if (failedTask?.errorDetails) {
+            onCopyTextButtonPress && onCopyTextButtonPress(failedTask?.errorDetails);
+        }
+    };
+
+    const handleRetryButtonPress = () => {
+        if (failedTask) {
+            onRetryButtonPress && onRetryButtonPress(failedTask);
+        }
+    };
+
+    const toggleExpand = () => {
+        setIsExpanded(!isExpanded);
+        LayoutAnimation.configureNext(animationConfig);
+    };
+
+    useEffect(() => {
+        if (taskHasError && failedTask) {
+            setIsExpanded(true);
+            console.error(
+                `Error in task ${failedTask.title}: ${failedTask.errorDetails?.message}`,
+                failedTask.errorDetails,
+            );
+        }
+    }, [taskHasError, failedTask]);
+
+    const renderItemText = () => (
+        <View>
+            <Typography
+                bold={taskStatus !== "success"}
+                color={taskStatus === "notStarted" ? "textDisabled" : "textPrimary"}
+                variant="body_short"
+                group="paragraph"
+            >
+                {title}
+            </Typography>
+
+            {description ? (
+                <View style={styles.descriptionContainer}>
+                    {taskStatus !== "notStarted" && taskCounter > 0 ? (
+                        <Typography>
+                            {completedTaskCounter} / {taskCounter} {description}
+                        </Typography>
+                    ) : (
+                        <Typography>{description}</Typography>
```

Comment:
Did the designs say something about the color of the description text if the task is not started? I see that the title is set to tertiary, but the description remains the same for these types of items.